src/Core/Managed/Shared/Channel/Transmission.cs

```diff
@@ -148,27 +153,39 @@ public string Id
                     return null;
                 }
 #else
-                Task timeoutTask = TaskEx.Delay(this.Timeout);
-
                 WebRequest request = this.CreateRequest(this.EndpointAddress);
                 Task<HttpWebResponseWrapper> sendTask = this.SendRequestAsync(request);
-                                
-                Task completedTask = await TaskEx.WhenAny(timeoutTask, sendTask).ConfigureAwait(false);
-                if (completedTask == timeoutTask)
+                Task timeoutTask = TaskEx.Delay(this.Timeout).ContinueWith(task =>
                 {
-                    request.Abort(); // And force the sendTask to throw WebException.
-                }
+                    if (!sendTask.IsCompleted)
+                    {
+                        request.Abort(); // And force the sendTask to throw WebException.
```

Comment:
There is a race condition here on sendTask completing after the check. I do not recall what happens if Abort is called on a finished request. If nothing, this is benign. To avoid this you can use a synchronous continuation to the send task and null out the request reference or set an interlocked flag. And yes - the issue existed before the change. :)